csrc/scheduler/registry.cpp

```diff
@@ -1720,6 +1740,13 @@ class PointWiseScheduler : public SchedulerEntry {
       return !reason.empty();
     }
 
+    // Reject if there is a non-trivial reduction
```

Comment:
Why is this here?

Reply:
These checks get run by `FusionKernelRuntime` when we have previously accepted a Fusion and need to check it against new inputs. If we first run with size-zero reductions but then pass inputs that make those reductions non-trivial, we need Pointwise and NoOp to reject. Same goes for the other direction. All these checks are avoided if we handle reductions at concretization instead since we can then assume the reduction is nontrivial in both concrete and reduction domains.

Reply:
Ah, sorry, I thought this was added the transpose scheduler since I saw `TransposeScheduler` at around line 1735.

Reply:
If we made size==0 special like size==1 then we wouldn't need this check.